hundred_x/client.py

```diff
@@ -308,12 +324,23 @@ def get_candlestick(self, symbol: str, **kwargs) -> Any:
             params=params,
         ).json()
 
-    def get_symbol(self, symbol: str) -> Any:
+    def get_symbol(self, symbol: str = None) -> Any:
         """
         Get the details of a specific symbol.
+        If symbol is None, return all symbols.
         """
-        endpoint = f"/v1/ticker/24hr?symbol={symbol}"
-        return requests.get(self.rest_url + endpoint).json()[0]
+        response = self.send_message_to_endpoint(
+            endpoint="/v1/ticker/24hr",
+            method="GET",
+            message={},
+            params={"symbol": symbol} if symbol else {},
```

Comment:
if we just use our wrapper object, we now return the async future which can be awaited in the client side.